process/block/baseProcess.go

```diff
@@ -3304,3 +3304,85 @@ func (bp *baseProcessor) getCurrentBlockHeader() data.HeaderHandler {
 
 	return bp.blockChain.GetGenesisHeader()
 }
+
+func (bp *baseProcessor) checkContextBeforeExecution(header data.HeaderHandler) error {
+	lastCommittedRootHash, err := bp.accountsDB[state.UserAccountsState].RootHash()
+	if err != nil {
+		return err
+	}
+
+	// TODO: the GetLastExecutedBlockInfo should return also the LastCommittedBlockInfo (in case the committed block was V2)
+	// this is done on another PR
+	lastExecutedNonce, lastExecutedHash, lastExecutedRootHash := bp.blockChain.GetLastExecutedBlockInfo()
+	if !bytes.Equal(header.GetPrevHash(), lastExecutedHash) {
+		return process.ErrBlockHashDoesNotMatch
+	}
+	if header.GetNonce() != lastExecutedNonce+1 {
```

Comment:
the newly created header will have the nonce based on last commited block + 1, right? it is ok to check it with last executed nonce?